src/libraries/Microsoft.PowerFx.Core/Public/Values/CustomObjectValue.cs

```diff
@@ -11,47 +11,41 @@
 
 namespace Microsoft.PowerFx.Core.Public.Values
 {
+    /// <summary>
+    /// The backing implementation for CustomObjectValue, for example Json, Xml,
+    /// or the Ast or Value system from another language.
+    /// </summary>
     public interface ICustomObject
     {
-        bool IsArray { get; }
-
-        bool IsNull { get; }
-
-        bool IsObject { get; }
-
-        bool IsString { get; }
-
-        bool IsNumber { get; }
-
-        object ToObject();
+        FormulaType Type { get; } // Use ExternalType if the type is incompatible with PowerFx
```

Comment:
>// [](http://example.com/codeflow?start=34&length=2)

nit: these should be xml comments ///  (since they're on a public interface) rather than just //. The C# IDE will then show xml comments as tooltips when called.

Reply:
Done